pkgs/development/tools/jq/default.nix

```diff
@@ -12,7 +13,16 @@ stdenv.mkDerivation rec {
 
   outputs = [ "bin" "doc" "man" "dev" "lib" "out" ];
 
-  buildInputs = [ oniguruma ];
+  # inspired by app-misc/jq/jq-1.6-r3.ebuild from Gentoo
+  postPatch = ''
+    rm -r ./modules/oniguruma
+  '' + lib.optionalString (!onigurumaSupport) ''
+    sed -e '/modules\/oniguruma/d' -i Makefile.am
+    sed -e 's,tests/onigtest,,' -i Makefile.am
+  '';
+
+  buildInputs = lib.optionals onigurumaSupport [ oniguruma ];
+  nativeBuildInputs = lib.optionals (!onigurumaSupport) [ autoreconfHook ];
```

Comment:
Please run `autoreconfHook` no matter what. It will make it easier to troubleshoot down the road if there's ever an issue with the build.